examples/ca_handler/mswcce_ca_handler.py

```diff
@@ -0,0 +1,186 @@
+
+# pylint: disable=E0401
+from acme_srv.helper import (
+    load_config,
+    b64_url_recode,
+    convert_byte_to_string,
+    convert_string_to_byte,
+    proxy_check,
+)
+
+
+class CAhandler(object):
+    """MS-WCCE CA handler"""
+
+    def __init__(self, _debug=None, logger=None):
+        self.logger = logger
+        self.host = None
+        self.user = None
+        self.password = None
+        self.template = None
+        self.proxy = None
+        self.target_domain = None
+        self.domain_controller = None
+        self.ca_name = None
+
+    def __enter__(self):
+        """Makes CAhandler a Context Manager"""
+        if not self.host:
+            self._config_load()
+        return self
+
+    def __exit__(self, *args):
+        """close the connection at the end of the context"""
+
+    def enroll(self, csr):
+        """enroll certificate via MS-WCCE"""
+        self.logger.debug("CAhandler.enroll({0})".format(self.template))
+        cert_bundle = None
+        error = None
+        cert_raw = None
+
+        if not (self.host and self.user and self.password and self.template):
+            self.logger.error("Config incomplete")
+            return ("Config incomplete", None, None, None)
+
+        target = Target(
+            domain=self.target_domain,
+            username=self.user,
+            password=self.password,
+            remote_name=self.host,
+            dc_ip=self.domain_controller,
+        )
+        request = Request(
+            target=target,
+            ca=self.ca_name,
+            template=self.template,
+        )
+
+        # recode csr
+        csr = textwrap.fill(b64_url_recode(self.logger, csr), 64) + "\n"
+
+        # TODO: currently getting certificate chain is not supported
+        ca_pem = ""
+
+        try:
+            # request certificate
+            cert_raw = convert_byte_to_string(
+                request.get_cert(convert_string_to_byte(csr))
+            )
+            # replace crlf with lf
+            cert_raw = cert_raw.replace("\r\n", "\n")
+        except Exception as err_:
+            cert_raw = None
+            self.logger.error(
+                "ca_server.get_cert() failed with error: {0}".format(err_)
+            )
+
+        if cert_raw:
+            cert_bundle = cert_raw + ca_pem
+            cert_raw = cert_raw.replace("-----BEGIN CERTIFICATE-----\n", "")
+            cert_raw = cert_raw.replace("-----END CERTIFICATE-----\n", "")
+            cert_raw = cert_raw.replace("\n", "")
+        else:
+            self.logger.error("cert bundling failed")
+            error = "cert bundling failed"
+
+        self.logger.debug("Certificate.enroll() ended")
+        return (error, cert_bundle, cert_raw, None)
+
+    def _config_load(self):
+        """ " load config from file"""
+        self.logger.debug("CAhandler._config_load()")
+        config_dic = load_config(self.logger, "CAhandler")
+        if config_ca_handler := config_dic.get("CAhandler"):
```

Comment:
Getting error here: Helper.ca_handler_load(): loading CAhandler configured in cfg failed with err: invalid syntax (mswcce_ca_handler.py, line 106)